Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -101,141 +76,50 @@ void ObservePreferenceChange (PreferenceChange change)
                 enabled = Prefs.Telemetry.Enabled.GetValue ();
                 if (!enabled) {
                     Log.Info (TAG, "Telemetry disabled.");
-                    httpClient = null;
-                } else if (httpClient == null) {
+                    appInsightsClient = null;
+                } else if (appInsightsClient == null) {
                     Log.Info (TAG, "Telemetry will be enabled after restart.");
                 }
             }
         }
 
-        public void Post (ITelemetryEvent evnt)
+        public void Post (Exception exception)
```

Comment:
Have this take a `LogEntry` instead so you can attach logging properties to it, particularly the message, which can and often does differ from the message _in the exception_.

Reply:
Great idea! Especially since the exceptions are somewhat truncated currently in Insights.